fuzz/fuzz_targets/int_set_op_processor.rs

```diff
@@ -0,0 +1,1045 @@
+use std::cmp::max;
+
+use std::cmp::min;
+use std::fmt::Debug;
+use std::io::Cursor;
+use std::io::Read;
+use std::ops::Bound::Excluded;
+use std::ops::Bound::Included;
+use std::ops::RangeInclusive;
+use std::{collections::BTreeSet, error::Error};
+
+use int_set::Domain;
+use int_set::InDomain;
+use int_set::IntSet;
+
+#[derive(PartialEq, Clone, Copy)]
+pub enum OperationSet {
+    Standard,
+    #[allow(dead_code)]
+    SparseBitSetEncoding,
+}
+
+pub trait SetMember<T>: Domain<T> + Ord + Copy + Debug {
+    fn create(val: u32) -> Option<T>;
+    fn can_be_inverted() -> bool;
+    fn increment(&mut self);
+}
+
+impl SetMember<u32> for u32 {
+    fn create(val: u32) -> Option<u32> {
+        Some(val)
+    }
+
+    fn can_be_inverted() -> bool {
+        false
+    }
+
+    fn increment(&mut self) {
+        *self = self.saturating_add(1);
+    }
+}
+
+/// This is an integer in the domain of [0, 2048). It's used by the fuzzer
+/// for testing inverted sets to avoid causing excessively long running operations
+/// and memory usage on the btree set kept along side the IntSet.
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord)]
+pub struct SmallInt(u32);
+
+impl SmallInt {
+    const MAX_VALUE: u32 = 4 * 512 - 1;
+
+    fn new(value: u32) -> SmallInt {
+        if value > Self::MAX_VALUE {
+            panic!("Constructed SmallInt with value > MAX_VALUE");
+        }
+        SmallInt(value)
+    }
+}
+
+impl SetMember<SmallInt> for SmallInt {
+    fn create(val: u32) -> Option<SmallInt> {
+        if val > Self::MAX_VALUE {
+            return None;
+        }
+        Some(SmallInt::new(val))
+    }
+
+    fn can_be_inverted() -> bool {
+        true
+    }
+
+    fn increment(&mut self) {
+        self.0 = min(self.0 + 1, Self::MAX_VALUE);
```

Comment:
in general there's no need to import `min` for this, you can just do `Self::MAX_VALUE.min(self.0 + 1)`.